_posts/2017-02-09-rapid-mysql-imports-docker.md

```diff
@@ -0,0 +1,186 @@
+
+``` bash
+pv /path/to/database.sql | mysql -u{some_user} -p {some_pass} {database_name}
+```
+
+On large databases, though, this can be a slow process. If we look at a database dump SQL file, we can see why. For example, a 19 MB database dump file I am using in one of my test cases further on in this post contains these instructions:
+
+``` sql
+--
+-- Table structure for table `block_content`
+--
+
+DROP TABLE IF EXISTS `block_content`;
+/*!40101 SET @saved_cs_client     = @@character_set_client */;
+/*!40101 SET character_set_client = utf8 */;
+CREATE TABLE `block_content` (
+  `id` int(10) unsigned NOT NULL AUTO_INCREMENT,
+  `revision_id` int(10) unsigned DEFAULT NULL,
+  `type` varchar(32) CHARACTER SET ascii NOT NULL COMMENT 'The ID of the target entity.',
+  `uuid` varchar(128) CHARACTER SET ascii NOT NULL,
+  `langcode` varchar(12) CHARACTER SET ascii NOT NULL,
+  PRIMARY KEY (`id`),
+  UNIQUE KEY `block_content_field__uuid__value` (`uuid`),
+  UNIQUE KEY `block_content__revision_id` (`revision_id`),
+  KEY `block_content_field__type__target_id` (`type`)
+) ENGINE=InnoDB AUTO_INCREMENT=12 DEFAULT CHARSET=utf8mb4 COMMENT='The base table for block_content entities.';
+/*!40101 SET character_set_client = @saved_cs_client */;
+
+--
+-- Dumping data for table `block_content`
+--
+
+LOCK TABLES `block_content` WRITE;
+/*!40000 ALTER TABLE `block_content` DISABLE KEYS */;
+set autocommit=0;
+INSERT INTO `block_content` VALUES (1,1,'basic','a9167ea6-c6b7-48a1-ac06-6d04a67a5d54','en'),(2,2,'basic','2114eee9-1674-4873-8800-aaf06aaf9773','en'),(3,3,'basic','855c13ba-689e-40fd-9b00-d7e3dd7998ae','en'),(4,4,'basic','8c68671b-715e-457d-a497-2d38c1562f67','en'),(5,5,'basic','bc7701dd-b31c-45a6-9f96-48b0b91c7fa2','en'),(6,6,'basic','d8e23385-5bda-41da-8e1f-ba60fc25c1dc','en'),(7,7,'basic','ea6a93eb-b0c3-4d1c-8690-c16b3c52b3f1','en'),(8,8,'basic','3d314051-567f-4e74-aae4-a8b076603e44','en'),(9,9,'basic','2ef5ae05-6819-4571-8872-4d994ae793ef','en'),(10,10,'basic','3deaa1a9-4144-43cc-9a3d-aeb635dfc2ca','en'),(11,11,'basic','d57e81e8-c613-45be-b1d5-5844ba15413c','en');
+/*!40000 ALTER TABLE `block_content` ENABLE KEYS */;
+UNLOCK TABLES;
+commit;
+```
+
+When I pipe the contents of the MySQL database dump to the `mysql` command, the `mysql` client processes each of these instructions sequentially in order to (1) create the structure for each table defined in the file and (2) populate the database with data from the SQL dump.
+
+What happens when `mysql` finished processing the SQL dump? The database contents live in `/var/lib/mysql/{database}`, so for example for the `block_content` table mentioned above there are two files called `block_content.frm` and `block_content.ibd` in `/var/lib/mysql/{database}/`. The `/var/lib/mysql` directory will also contain a number of other directories and files related to the configuration of the MySQL server.
```

Comment:
For me, this paragraph doesn't further express the "sequential-ness" of the import process and how long it takes. To me, it's just saying that these files are in this directly after the import. So maybe consider removing it? Or describing clearly how those 2 files interact to slow things down?